app/test2.py

```diff
@@ -0,0 +1,28 @@
+def factorial(n):
+    result = 1
+    for i in range(2, n + 1):
+        result *= i
+    return result
+
+
+def is_prime(n):
+    if n <= 1:
```

Comment:
Enhance the `is_prime` function's error handling by adding checks for non-integer and negative inputs, ensuring the function gracefully handles unexpected input types and values.

**Label**: robustness

**Code suggestion**

```python
def is_prime(n):
    if not isinstance(n, int) or n <= 1:
        return False
```